src/main/java/edu/wpi/grip/ui/pipeline/input/NumberSpinnerInputSocketView.java

```diff
@@ -0,0 +1,66 @@
+package edu.wpi.grip.ui.pipeline.input;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.InvalidationListener;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.SpinnerValueFactory;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * An {@link InputSocketView} that lets the user set the value of a number with a {@link Spinner}
+ */
+public class NumberSpinnerInputSocketView extends InputSocketView<Number> {
+
+    private final static Object[] DEFAULT_DOMAIN = new Object[]{Double.MIN_VALUE, Double.MAX_VALUE};
```

Comment:
You are aware that `Double.MIN_VALUE` is a positive number http://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#MIN_VALUE
If you want the lowest value for a Double you want `-Double.MAX_VALUE`